timemachine/cpp/src/kernels/k_exchange.cu

```diff
@@ -59,76 +65,47 @@ void __global__ k_setup_proposals(
     }
 }
 
-template <typename RealType>
-void __global__ k_attempt_exchange_move(
-    const int N,
-    const RealType *__restrict__ rand,           // [1]
-    const RealType *__restrict__ before_max,     // [1]
-    const RealType *__restrict__ before_log_sum, // [1]
-    const RealType *__restrict__ after_max,      // [1]
-    const RealType *__restrict__ after_log_sum,  // [1]
-    const int *__restrict__ mol_offsets,         // [num_mols + 1]
-    const int *__restrict__ selected_mol,        // [1]
-    const double *__restrict__ moved_coords,     // [num_atoms_in_each_mol, 3]
-    double *__restrict__ dest_coords,            // [N, 3]
-    size_t *__restrict__ num_accepted            // [1]
+void __global__ k_accepted_exchange_move(
+    const int batch_size,
+    const int num_atoms_in_each_mol,
+    const int *__restrict__ accepted_batched_move, // [1]
+    const int *__restrict__ mol_idx_per_batch,     // [batch_size]
+    const int *__restrict__ mol_offsets,           // [num_target_mols]
+    const double *__restrict__ moved_coords,       // [batch_size, num_atoms_in_each_mol, 3]
+    double *__restrict__ dest_coords,              // [N, 3]
+    size_t *__restrict__ num_accepted,             // [1],
+    int *__restrict__ rand_offset                  // [1]
 ) {
     // Note that this kernel does not handle multiple proposals, expects that the proposals
```

Comment:
Is this comment still relevant now that it supports batching?

Reply:
Yes, this kernel just takes the output of `k_select_first_valid_move`